csharp/src/Apache.Arrow.Flight/Client/FlightClient.cs

```diff
@@ -32,35 +32,35 @@ public FlightClient(ChannelBase grpcChannel)
             _client = new FlightService.FlightServiceClient(grpcChannel);
         }
 
-        public AsyncServerStreamingCall<FlightInfo> ListFlights(FlightCriteria criteria = null, Metadata headers = null)
+        public AsyncServerStreamingCall<FlightInfo> ListFlights(FlightCriteria criteria = null, Metadata headers = null, System.DateTime? deadline = null, System.Threading.CancellationToken cancellationToken = default(global::System.Threading.CancellationToken))
```

Comment:
nit: consider `using System.Threading` unless it creates a conflict, and `default` instead of `default(CancellationToken)`.